data-prepper-plugins/mutate-string-processors/README.md

```diff
@@ -0,0 +1,148 @@
+# Mutate String Processors
+The following is a list of processors to mutate a string.
+
+## SubstituteStringProcessor
+A processor that matches a field value against a regular expression and replaces all matches with a replacement string.
+
+### Basic Usage
+To start using Substitute String Processor with Data Prepper, create the following `pipeline.yaml`.
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - substitute_string:
+        entries:
+          - source: "source"
+            from: "from"
+            to: "to"
+          - source: "source2"
+            from: "from2"
+            to: "to2"
+  sink:
+    - stdout:
+```
+
+### Configuration
+* `entries` - (required) - A list of entries to add to an event
+    * `source` - (required) - The entry to be modified
+    * `from` - (required) - The string to be replaced
+    * `to` - (required) - The string it will be substituted to
```

Comment:
I would recommend avoiding using pronouns in documentation. "It" in this context is referring to something other than the "to" field. Let's make it easier for the reader and be explicit. Here is a suggestion

```
the string to be substituted for each match of `from`
```